queries.sql

```diff
@@ -23,3 +23,57 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
 /*Animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)*/
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- Update and revert species column
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+-- Update species column with 'digimon' or 'pokemon'
+BEGIN;
+UPDATE animals SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon'
+WHERE name NOT LIKE '%mon';
+COMMIT;
+
+-- Delete all records then roll back transaction
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+-- Create a savepoint
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+SAVEPOINT before_2022;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO before_2022;
+
+UPDATE animals SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals
+WHERE escape_attempts = 0;
```

Comment:
- [ ] Let's add query to delete all records in the animal table and roll back the transaction